prow/tide/tide.go

```diff
@@ -173,6 +243,19 @@ func (c *Controller) syncSubpool(ctx context.Context, sp subpool) error {
 		}
 		presubmits = append(presubmits, ps.Name)
 	}
+	batchMerge, batchPending := accumulateBatch(presubmits, sp.prs, sp.pjs)
+	// Do not take any actions while waiting for a batch to complete. We don't
+	// want to invalidate the old batch result.
+	if batchPending {
+		c.log.Info("Waiting for batch to complete.")
+		return nil
+	}
+	if len(batchMerge) > 0 {
+		for _, pr := range batchMerge {
+			c.log.Infof("Merge PR #%d (batch).", int(pr.Number))
+		}
+		return nil
+	}
 	successes, pendings, nones := accumulate(presubmits, sp.prs, sp.pjs)
```

Comment:
Slightly related because of the order, `accumulate` returns PRs that have green tests but are not necessarily tested against HEAD in successes which are then merged. We don't want that, right?

Reply:
They will be tested against HEAD. `divideSubpool` removes all PJs that weren't tested against HEAD.

Reply:
ok